### PR TITLE
feat: add token security api

### PR DESCRIPTION
**Pull request by Randolph314**
## Description

Closes [#478](https://mask.atlassian.net/browse/MF-478)

## Type of change

- [ ] Documentation
- [ ] Code refactoring (Restructuring existing code w/o changing its observable behavior)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (a fix or feature that would make something no longer possible to do/require old user must upgrade their Mask Network to this new version)

## Checklist

- [ ] My code follows the style guidelines of this project.
- [ ] I have performed a self-review of my own code.
  - [ ] I have removed all in development `console.log`s
  - [ ] I have removed all commented code.
- [ ] I have commented on my code, particularly in hard-to-understand areas.
- [ ] I have read [Internationalization Guide](https://github.com/DimensionDev/Maskbook/blob/develop/docs/i18n-guide.md) and moved text fields to the i18n JSON file.

If this PR depends on external APIs:

- [ ] I have configured those APIs with CORS headers to let extension requests get passed. 
  - chrome extension: `chrome-extension://[id]`
  - firefox extension: `moz-extension://[id]`
- [ ] I have delegated all web requests to the background service via the internal RPC bridge.

### CODE REVIEW

File: cspell.json [hunk 1]
```diff
@@ -170,6 +170,7 @@
         "stego",
         "subrepo",
         "sushiswap",
+        "swaping",
```
Comment (Jack-Works) [inline cspell.json]: Do you mean `swapping`?

File: packages/shared/src/UI/components/TokenSecurity/rules.ts [hunk 2]
```diff
@@ -0,0 +1,325 @@
+import { SecurityMessageLevel } from './Common'
+import type { useSharedI18N } from '../../../locales'
+import parseInt from 'lodash-es/parseInt'
```
Comment (Jack-Works) [inline packages/shared/src/UI/components/TokenSecurity/rules.ts:3]:
```suggestion
import { parseInt } from 'lodash-unified'
```

File: packages/mask/src/plugins/Trader/SNSAdaptor/trending/TrendingViewDeck.tsx [hunk 3]
```diff
@@ -122,6 +126,19 @@ export function TrendingViewDeck(props: TrendingViewDeckProps) {
...
+    const isTokenSecurityClosed = snsAdaptorMinimalPlugins.map((x) => x.ID).includes(PluginId.GoPlusSecurity)
+    const { value: tokenSecurityInfo, error } = useAsyncRetry(async () => {
+        if (isTokenSecurityClosed) return
+        if (!coin?.contract_address) return
+        const values = await GoPlusLabs.getTokenSecurity(chainId, [coin.contract_address.trim()])
+        if (!Object.keys(values ?? {}).length) throw new Error('Contract Not Found')
+        return Object.entries(values ?? {}).map((x) => ({ ...x[1], contract: x[0], chainId }))[0] as
+            | TokenSecurity
+            | undefined
+    }, [chainId, coin, isTokenSecurityClosed])
```
Comment (Jack-Works) [inline packages/mask/src/plugins/Trader/SNSAdaptor/trending/TrendingViewDeck.tsx]: This code repeats in the PR. Move it to its own hook.

File: packages/polyfills/types/__all.d.ts [hunk 4]
```diff
@@ -1,2 +1,3 @@
 /// <reference path="./env.d.ts" />
 /// <reference path="./intl.d.ts" />
+/// <reference path="./theme-global.d.ts" />
```
Comment (Jack-Works) [inline packages/polyfills/types/__all.d.ts:3]: Don't add it here.

File: packages/.eslintrc.json [hunk 5]
```diff
@@ -40,8 +40,6 @@
         "no-unmodified-loop-condition": "error",
         "no-unneeded-ternary": "error",
         "no-useless-concat": "error",
-        "no-useless-rename": "error",
```
Comment (septs) [inline packages/.eslintrc.json:43]: plz DO NOT MODIFY `.eslintrc`

File: cspell.json [hunk 6]
```diff
@@ -120,7 +120,6 @@
         "multicall",
         "multihop",
         "newsfeed",
-        "nftavatar",
```
Comment (septs) [inline cspell.json:123]: plz keep sync develop version

File: packages/.eslintrc.json (hunk 5 above)
Comment (Jack-Works) [reply packages/.eslintrc.json:43]: @Randolph314 looks like you didn't sync correctly with the base branch. You're reverting many commits.

Comment (Randolph314) [reply packages/.eslintrc.json:43]: @Jack-Works so I need to merge develop branch?

File: package.json [hunk 7]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "mask-network",
-  "packageManager": "pnpm@7.1.1",
```
Comment (Jack-Works) [inline package.json:3]: Looks like you didn't merge the develop branch correctly. You're reverting many changes.

File: packages/shared/src/hooks/useTokenSecurity.ts [hunk 8]
```diff
@@ -0,0 +1,20 @@
...
+export type TokenSecurity = SecurityAPI.ContractSecurity &
+    SecurityAPI.TokenSecurity &
+    SecurityAPI.TradingSecurity & { contract: string; chainId: ChainId }
+
+export const useTokenSecurity = (chainId: ChainId, address?: string, isTokenSecurityClosed = false) => {
+    return useAsyncRetry(async () => {
+        if (isTokenSecurityClosed) return
+        if (!address) return
+        if (!address || address === ZERO_ADDRESS) return
+        const values = await GoPlusLabs.getTokenSecurity(chainId, [address])
```
Comment (septs) [inline packages/shared/src/hooks/useTokenSecurity.ts]:
```suggestion
        let values = await GoPlusLabs.getTokenSecurity(chainId, [address])
        values ??= {}
```

File: packages/shared/src/hooks/useTokenSecurity.ts [hunk 9]
```diff
@@ -0,0 +1,20 @@
...
+        if (!Object.keys(values ?? {}).length) throw new Error('Contract Not Found')
```
Comment (septs) [inline packages/shared/src/hooks/useTokenSecurity.ts]:
```suggestion
        if (isEmpty(values)) throw new Error('Contract Not Found')
```

https://lodash.com/docs/4.17.15#isEmpty

File: packages/shared/src/hooks/useTokenSecurity.ts [hunk 10]
```diff
@@ -0,0 +1,20 @@
...
+        return Object.entries(values ?? {}).map((x) => ({ ...x[1], contract: x[0], chainId }))[0] as
```
Comment (septs) [inline packages/shared/src/hooks/useTokenSecurity.ts]:
```suggestion
        const entity = first(Object.entries(values ?? {}))
        if (!entity) return
        return ({ ...entity[1], contract: entity[0], chainId }) as TokenSecurity
```

File: packages/plugins/GoPlusSecurity/src/SNSAdaptor/CheckSecurityDialog.tsx [hunk 11]
```diff
@@ -64,12 +65,13 @@ export function CheckSecurityDialog(props: BuyTokenDialogProps) {
...
-        const values = await GoPlusLabs.getTokenSecurity(chainId, [content.trim()])
-        if (!Object.keys(values ?? {}).length) throw new Error('Contract Not Found')
-        return Object.entries(values ?? {}).map((x) => ({ ...x[1], contract: x[0], chainId }))[0] as
-            | TokenSecurity
-            | undefined
+        if (!content || content.trim() === ZERO_ADDRESS) return
+        let values = await GoPlusLabs.getTokenSecurity(chainId, [content.trim()])
+        values ??= {}
+        if (isEmpty(values)) throw new Error('Contract Not Found')
+        const entity = first(Object.entries(values ?? {}))
```
Comment (septs) [inline packages/plugins/GoPlusSecurity/src/SNSAdaptor/CheckSecurityDialog.tsx:72]:
```suggestion
        const entity = first(Object.entries(values))
```

**Comment by septs** [comment]
@Randolph314 plz resolve conflicts

**Pull request closed without merging**